tests/test_permissions_cli.py

```diff
@@ -0,0 +1,500 @@
+                    ]
+                },
+                "deposit-update": {
+                    "roles": [],
+                    "users": ["info@inveniosoftware.org"]
+                }
+            }
+        },
+        status=201)
+
+    res = cli_run(
+        "permissions add -p some-pid -u info@inveniosoftware.org -r read")
+
+    assert responses.calls[0].request.headers[
+        'Accept'] == 'application/permissions+json'
+    assert responses.calls[0].request.headers[
+        'Content-Type'] == 'application/json'
+
+    assert json.loads(responses.calls[0].request.body) == [{
+        "email": "info@inveniosoftware.org",
+        "type": "user",
+        "op": "add",
+        "action": "deposit-read"
+    }]
+
+    assert res.exit_code == 0
+    assert json.loads(res.stripped_output) == {
+        "deposit-admin": {
+            "roles": [],
+            "users": ["info@inveniosoftware.org"]
+        },
+        "deposit-read": {
+            "roles": [],
+            "users": ["info@inveniosoftware.org", "cms@inveniosoftware.org"]
+        },
+        "deposit-update": {
+            "roles": [],
+            "users": ["info@inveniosoftware.org"]
+        }
+    }
+
+
+@responses.activate
+def test_permissions_add_egroup(cli_run):
+    responses.add(
+        responses.POST,
+        "https://analysispreservation-dev.cern.ch/api/deposits/some-pid/actions/permissions",
+        json={
+            'permissions': {
+                "deposit-admin": {
+                    "roles": [],
+                    "users": ["info@inveniosoftware.org"]
+                },
+                "deposit-read": {
+                    "roles": [],
+                    "users": [
+                        "info@inveniosoftware.org", "cms@inveniosoftware.org"
+                    ]
+                },
+                "deposit-update": {
+                    "roles": [],
+                    "users": ["info@inveniosoftware.org"]
+                }
+            }
+        },
+        status=201)
+
+    res = cli_run(
+        "permissions add -p some-pid -e info@inveniosoftware.org -r read")
+
+    assert responses.calls[0].request.headers[
+        'Accept'] == 'application/permissions+json'
+    assert responses.calls[0].request.headers[
+        'Content-Type'] == 'application/json'
+
+    assert json.loads(responses.calls[0].request.body) == [{
+        "email": "info@inveniosoftware.org",
+        "type": "egroup",
+        "op": "add",
+        "action": "deposit-read"
+    }]
+
+    assert res.exit_code == 0
+    assert json.loads(res.stripped_output) == {
+        "deposit-admin": {
+            "roles": [],
+            "users": ["info@inveniosoftware.org"]
+        },
+        "deposit-read": {
+            "roles": [],
+            "users": ["info@inveniosoftware.org", "cms@inveniosoftware.org"]
+        },
+        "deposit-update": {
+            "roles": [],
+            "users": ["info@inveniosoftware.org"]
+        }
+    }
```

Comment:
it shouldn't work like this
when you're adding an egroup permission, new entry will be added to `roles` not `users` (we map egroups to roles)